### PR TITLE
add add sanitizers to unit test cmake build

### PR DESCRIPTION
**Pull request by jvishnefske**
* currently enabled on 64 bit unit test build. (edit: sanitizers added to x32 build by beaf0a8)
* silently falls back if -fsanitize=undefined support is not detected
* verifies bug-203 fix

### CODE REVIEW

**Comment by jvishnefske** [comment]
I am attempting to build with and without including the FIX-204 to see if the issue shows up in the log. currently a sanitizer runtime warning does not cause the unit test to fail.

**Comment by jvishnefske** [comment]
I was able to get the cmake unit tests to fail from an actual undefined behavior on gcc. Clang currently has a link issue blocking this. I will have another go at it tomorrow.
the 32 bit version of the symbol needed to enable undefined sanitizer on clang appear to be avaliable as a static library on ubuntu docker. 

  user@80a9238469f6:/usr/lib32$ grep -r __mulodi4 /usr
  Binary file /usr/lib/x86_64-linux-gnu/libLLVM-12.so.1 matches
  Binary file /usr/lib/llvm-12/lib/clang/12.0.0/lib/linux/libclang_rt.builtins-i386.a matches
  Binary file /usr/lib/llvm-12/lib/clang/12.0.0/lib/linux/libclang_rt.builtins-x86_64.a matches

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -10,6 +10,9 @@ enable_testing()
 set(CTEST_OUTPUT_ON_FAILURE ON)
 set(library_dir "${CMAKE_SOURCE_DIR}/../libcanard")
 
+set(NO_STATIC_ANALYSIS OFF CACHE BOOL "disable canard static analysis") 
+set(NO_CANARD_SANATIZER OFF CACHE BOOL "disable canard runtime sanitizers")
```
Comment (pavel-kirienko) [inline tests/CMakeLists.txt:14]: Rather than stopping halfway I suggest you go forth and call it SATANIZER :imp:

Comment (jvishnefske) [reply tests/CMakeLists.txt]: updated spelling on bdd90136ba9acd6737ea5dba821f74c379c0f865

**Comment by pavel-kirienko** [comment]
Please tag me when ready for review.

**Comment by jvishnefske** [comment]
@pavel-kirienko the basic concept here seems to be working, and was able to fail on the #203 issue in my local tests. I considered adding support with toolchain files, however that would have been (possbily)even more invasive since the entire build matrix would have to be broken into several toolchain files. This implementation uses runtime compile checks to query the compiler for basic support, and then sets the resulting binary to fail if any of a list of checks are encountered. The disadvantage of this approach is that compilers which are new enough to support -fsanitize=undefined, but do not support one of the long list of "fatal" checks may have compile errors. since the cmake build is recomended to only be used for unit tests. I don't know if this is a disadvantage for any use case.

**Review by pavel-kirienko — APPROVED** [review]

**Pull request merged**